src/System.Net.Http/src/System/Net/Http/OSX/CurlHandler.SslProvider.cs

```diff
@@ -0,0 +1,133 @@
+                {
+                    // libcurl (as of 7.49.1) does not have any callback which can be registered which fires
+                    // between the time that a TLS/SSL handshake has offered up the server certificate and the
+                    // time that the HTTP request headers are written.  Were there any callback, the option
+                    // CURLINFO_TLS_SSL_PTR could be queried (and the backend identifier validated to be
+                    // CURLSSLBACKEND_DARWINSSL). Then the SecTrustRef could be extracted to build the chain,
+                    // a la SslStream.
+                    //
+                    // Without the callback the matrix looks like:
+                    // * If default-trusted and callback-would-trust: No difference (except side effects, like logging).
+                    // * If default-trusted and callback-would-block: Data would have been sent in violation of user trust.
+                    // * If not-default-trusted and callback-would-not-trust: No difference (except side effects).
+                    // * If not-default-trusted and callback-would-trust: No data sent, which doesn't match user desires.
+                    //
+                    // Of the two "different" cases, sending when we shouldn't is worse, so that's the direction we
+                    // have to cater to. So we'll use default trust, and throw on any custom callback.
+                    //
+                    // The situation where system trust fails can be remedied by including the certificate into the
+                    // user's keychain and setting the SSL policy trust for it to "Always Trust".
+                    // Similarly, the "block this" could be attained by setting the SSL policy for a cert in the
+                    // keychain to "Never Trust".
+                    throw new PlatformNotSupportedException(
+                        SR.Format(
+                            SR.net_http_libcurl_callback_notsupported,
+                            CurlVersionDescription,
+                            CurlSslVersionDescription));
+                }
+
+                // Revocation checking is always on for darwinssl (SecureTransport).
+                // If any other backend is used and revocation is requested, we can't guarantee
+                // that assertion.
+                if (easy._handler.CheckCertificateRevocationList &&
+                    !CurlSslVersionDescription.Equals("SecureTransport"))
+                {
+                    throw new PlatformNotSupportedException(
+                    SR.Format(
+                        SR.net_http_libcurl_revocation_notsupported,
+                        CurlVersionDescription,
+                        CurlSslVersionDescription));
+                }
+
+                SetSslVersion(easy);
+            }
+
+            private static void SetSslVersion(EasyRequest easy)
```

Comment:
This looks the same as for the Unix version (I think).  Are they the same?  If so, could we factor this out into a partial file that's shared with both versions?